### PR TITLE
ref: Replace last LRU caches with moka

### PR DESCRIPTION
**Pull request by loewenheim**

### CODE REVIEW

File: crates/symbolicator-service/src/services/download/gcs.rs [hunk 1]
```diff
@@ -41,7 +42,7 @@ impl GcsDownloader {
     /// If the cache contains a valid token, then this token is returned. Otherwise, a new token is
     /// requested from GCS and stored in the cache.
     async fn get_token(&self, source_key: &Arc<GcsSourceKey>) -> Result<Arc<GcsToken>, GcsError> {
-        if let Some(token) = self.token_cache.lock().unwrap().get(source_key) {
+        if let Some(token) = self.token_cache.get(source_key) {
```
[the remark this answers is not in the document]
Comment (loewenheim) [reply crates/symbolicator-service/src/services/download/gcs.rs:45]: Yep, you're completely right

File: crates/symbolicator-service/src/services/download/sentry.rs [hunk 2]
```diff
@@ -32,13 +32,12 @@ struct SearchQuery {
 }
 
 /// An LRU cache sentry DIF index responses.
-type SentryIndexCache = lru::LruCache<SearchQuery, (Instant, Vec<SearchResult>)>;
+type SentryIndexCache = moka::future::Cache<SearchQuery, Vec<SearchResult>>;
```
[the remark this answers is not in the document]
Comment (loewenheim) [reply crates/symbolicator-service/src/services/download/sentry.rs]: `SearchResult` is effectively a `String`, which would need to be cloned in `list_files` anyway. Not sure it makes sense to change this.

File: crates/symbolicator-service/src/services/download/sentry.rs [hunk 3]
```diff
@@ -32,13 +32,12 @@ struct SearchQuery {
 }
 
 /// An LRU cache sentry DIF index responses.
-type SentryIndexCache = lru::LruCache<SearchQuery, (Instant, Vec<SearchResult>)>;
+type SentryIndexCache = moka::future::Cache<SearchQuery, Result<Vec<SearchResult>, CacheError>>;
```
Comment (Swatinem) [inline crates/symbolicator-service/src/services/download/sentry.rs:35]:
```suggestion
type SentryIndexCache = moka::future::Cache<SearchQuery, CacheEntry<Vec<SearchResult>>>;
```

File: crates/symbolicator-service/src/services/download/sentry.rs [hunk 4]
```diff
@@ -98,37 +97,25 @@ impl SentryDownloader {
...
-            if created.elapsed() < self.cache_duration {
-                return Ok(entries.clone());
-            }
-        }
-
-        tracing::debug!(
-            "Fetching list of Sentry debug files from {}",
-            &query.index_url
-        );
+        let query_ = query.clone();
```
Comment (loewenheim) [inline crates/symbolicator-service/src/services/download/sentry.rs:100]: @Swatinem Do you see a way to avoid this clone under a new name?

Comment (Swatinem) [reply crates/symbolicator-service/src/services/download/sentry.rs:100]: not really, except creating a block just to create the future, though I wanted to explicitly avoid excessive rightward drift

**Review by Swatinem — APPROVED** [review]

**Pull request merged**